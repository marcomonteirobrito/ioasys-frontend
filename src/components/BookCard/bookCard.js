import React from 'react';
import PropTypes from 'prop-types';

import {
  Container,
  ImageContainer,
  BookImage,
  DetailContainer,
  DetailHeader,
  Title,
  Authors,
  DetailBody,
  DetailSpan,
} from './styles';

const BookCard = ({ booksData, idBookCard }) => (
  <Container id={idBookCard}>
    <ImageContainer>
      <BookImage src={booksData.imageURL} />
    </ImageContainer>
    <DetailContainer>
      <DetailHeader>
        <Title>{booksData.title}</Title>
        {booksData.authors?.map((author) => (
          <Authors>{author}</Authors>
        ))}
      </DetailHeader>

      <DetailBody>
        <DetailSpan>{booksData.pageCount}</DetailSpan>
        <DetailSpan>{booksData.publisher}</DetailSpan>
        <DetailSpan>{booksData.published}</DetailSpan>
      </DetailBody>
    </DetailContainer>
  </Container>
);

BookCard.propTypes = {
  booksData: PropTypes.objectOf(PropTypes.object).isRequired,
  idBookCard: PropTypes.string,
};

BookCard.defaultProps = {
  idBookCard: 'book-card',
};

export default BookCard;
