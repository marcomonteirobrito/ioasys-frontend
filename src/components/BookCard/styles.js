import styled from 'styled-components';

// import Teste from '../../assets/teste.jpg';
import Colors from '../../commons/Colors';

export const Container = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;

  background: #ffff;
  border-radius: 4px;
  width: 272px;
  height: 160px;

  cursor: pointer;
`;
export const BookContainer = styled.div`
  width: 224px;
  height: 122px;
  display: flex;
  justify-content: center;
  align-items: center;
`;
export const BookImage = styled.div`
  width: 81px;
  height: 122px;
`;
export const BookDetail = styled.div`
  display: flex;
  align-items: flex-start;
  justify-content: flex-start;
  flex-direction: column;

  width: 143px;
  max-height: 122px;
  margin-left: 16px;
`;
export const BookTitle = styled.span`
  max-width: 127px !important;
  height: 20px;
  color: ${Colors.black};
  margin-bottom: 2px;
`;
export const BookAuthor = styled.span`
  color: ${Colors.authorBookDetail};
  width: 127px;
  height: 20px;
`;
export const BookInformations = styled.div`
  height: 60px;
  width: 104px;
  display: flex;
  flex-direction: column;
  margin-top: 28px;
`;
export const BookSpan = styled.span`
  color: ${Colors.detailBook};
`;
export const DetailSpan = styled.span``;
