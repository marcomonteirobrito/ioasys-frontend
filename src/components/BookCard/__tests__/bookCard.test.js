/* eslint-disable no-undef */
import React from 'react';
import { render } from '@testing-library/react';

import BookCard from '../BookCard';

const bookDataMock = [
  {
    authors: ['Raul Pereira', 'Júlia Pereira Filho', 'Natália Barros Jr.'],
    title: 'Assumenda',
    description:
      'Est cupiditate rerum sed nisi quam dolorem vel qui veniam. Veritatis quam expedita. Dolor unde animi est dolor unde qui nisi. Iusto in saepe repellendus officia omnis aut reiciendis sapiente. Voluptate et at ut. Velit est iure repellat doloribus omnis earum laborum.\n \rModi sapiente laudantium similique recusandae modi eveniet minima sint in. Eum autem nihil est enim a nostrum quaerat magnam. Et tempora iure quidem aliquid rerum accusamus id.',
    pageCount: 990,
    category: 'Biografias',
    imageUrl: 'https://files-books.ioasys.com.br/Book-0.jpg',
    language: 'Inglês',
    isbn10: '6163701883',
    isbn13: '951-6163701883',
    publisher: 'Barros - Xavier',
    published: 2003,
    id: '60171639faf5de22b804a06e',
  },
  {
    authors: ['Lorraine Barros', 'Sra. Ofélia Silva', 'Warley Pereira'],
    title: 'Aut',
    description:
      'Qui omnis est commodi consectetur est. Odio ullam id ipsam suscipit numquam iusto iste consectetur. Sed ut est. Qui repellat corrupti veritatis consequatur quos ipsa.\n \rAtque dignissimos et eligendi numquam placeat ex. Consectetur sed aut autem ea id suscipit nihil a. Sunt repellendus suscipit reiciendis quidem sequi aut optio placeat quia. Quidem excepturi vel voluptate nisi molestiae soluta. Et nam et impedit illo accusamus sed earum est recusandae. Et delectus repellat enim distinctio velit et.',
    pageCount: 965,
    category: 'Biografias',
    imageUrl: 'https://files-books.ioasys.com.br/Book-5.jpg',
    language: 'Português',
    isbn10: '3355205900',
    isbn13: '112-3355205900',
    publisher: 'Martins - Xavier',
    published: 2019,
    id: '60171639faf5de22b804a16c',
  },
];

const renderBookCard = () => render(<BookCard bookData={bookDataMock} />);

describe('Input', () => {
  it('Deve renderizar o componente corretamente', () => {
    renderBookCard();

    expect(document.body).toMatchSnapshot();
  });
});
