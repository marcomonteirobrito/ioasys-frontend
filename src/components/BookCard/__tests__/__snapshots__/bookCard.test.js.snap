// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Deve renderizar o componente corretamente 1`] = `
<body>
  <div>
    <div
      class="sc-bdfBwQ dIRzjT"
      data-testid="bookContainer"
      id="book-card"
    >
      <div
        class="sc-gsTCUz iKaSyU"
      >
        <div
          class="sc-dlfnbm kumDvM"
        >
          teste
        </div>
        <div
          class="sc-hKgILt eqSQca"
        >
          <span
            class="sc-eCssSg jVTndr"
          />
          <span
            class="sc-jSgupP kqTHcr"
          />
          <div
            class="sc-gKsewC egdnxP"
          >
            <span
              class="sc-iBPRYJ dPYJaO"
            />
            <span
              class="sc-iBPRYJ dPYJaO"
            />
            <span
              class="sc-iBPRYJ dPYJaO"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
